.github/workflows/preview.yml

```diff
@@ -1,20 +1,51 @@
-name: Surge PR Preview
+name: PR Preview
 
 on:
+  push:
   pull_request:
-    paths-ignore:
-      - '**.md'
-      - '.all-contributorsrc'
+    types:
+      - opened
+      - synchronize
 
 jobs:
-  preview:
+
+  # This job will:
+  # -deploy a draft every time there is a pull request created or synchronized that is not on master branch
+  # -comment on that pull request with the deploy URL
+  deployPRDraft:
+    name: Deploy draft to Netlify
     runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request' && github.ref != 'refs/heads/master'
     steps:
-      - uses: actions/checkout@v2
-      - uses: afc163/surge-preview@v1
+      - name: Checkout repository
+        uses: actions/checkout@v1
+
+      - name: Deploy draft to Netlify
+        uses: South-Paw/action-netlify-deploy@v1.0.4
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          netlify-auth-token: ${{ secrets.NETLIFY_AUTH_TOKEN }}
+          netlify-site-id: ${{ secrets.NETLIFY_SITE_ID }}
+          build-dir: '.'
+          draft: true
+          comment-on-pull-request: true
+
+  # This job will:
+  # -deploy a production build every time there is a push on the master branch
+  # -comment that commit with the deploy URL
+  publishMasterCommit:
```

Comment:
Is this on pushing to the master branch? We don't need an action for that as we are using github pages for the moment, but please comment it out in case we need it later on.

Reply:
Ok. So, do I need to change the action for every PR to master branch too?

Reply:
We just need the action to run when a PR is submitted. It shouldn't run on merge or push to master